Scripts/DCS-BIOS/lib/modules/aircraft_modules/F-15E.lua

```diff
@@ -1171,4 +1171,597 @@ F_15E:defineMultipositionSwitch("R_LCP_LASER_DIG_4", 44, 3073, 73, 8, 0.1, "Rear
 F_15E:defineReadWriteRadio("UHF_RADIO", 7, 7, 3, 1000, "UHF RADIO")
 F_15E:defineReadWriteRadio("ARC_210_RADIO", 8, 7, 3, 1000, "ARC-210 RADIO")
 
+local F_MPDL_BUTTONS = {}
+
+local F_MPCD_BUTTONS = {}
+
+local F_MPDR_BUTTONS = {}
+
+local function split(stringvalue, delimiter)
+    local result = {};
+    for match in (stringvalue .. delimiter):gmatch("(.-)" .. delimiter) do
+        table.insert(result, match);
+    end
+    return result;
+end
+
+--[[ types PB entries may be missing ; write entries only if they're nil,  "`" or "^" ; add child values with $ for splitting ; only acceptable suffix is _LABEL ;
+PB%d 							PB1 to PB9
+PB%d%d 							PB01 to PB20 or PB10 to PB20
+[random prefix]_PB%d%d_LABEL 	PACS_PB01-20_LABEL
+PB%d%d-%d%d						PB01-02 for split on PB01 and PB02
+PB%d%d_PB%d%d					PB12_PB13 for split on PB12 and PB13
+[any prefix]_PB%d				TFR_PB5
+[any prefix]_PB%d%d				TFR_PB15 or TPOD_PB01 to TPOD_PB20 or SWPN_PB04
+PB%d%d_Value					is data begins 3 lines after "children are {" line
+PB%d%d%d%d						PB0405 for split between 04 and 05
+PB%d%d_%d%d						PB13_14 for split between 13 and 14
+]]--
+
+local function parseMFDButtons(stringindication)
+	local input = split(stringindication, "%c")
+	local values = {};
+	for numline, checkline in pairs(input) do
+		if checkline  and string.sub(checkline, 1, 1) == "-" then
+			local start = numline+1
+			local val = input[start]
+			local double = false
+			if val then
+				local pos = string.match(val, "^PB(%d[%d]?)$")
+				if not pos then
+					pos = string.match(val, "^[%a+_]+PB(%d+)[_LABEL]-$")
+					if not pos then
+						pos = string.match(val, "^PB(%d+)%-%d+$")
+						if not pos then
+							pos = string.match(val, "^PB(%d%d)[%d%d|_PB%d%d|_%d%d]+$")
+							double = true
+						else
+							double = true
+						end
+					end
+				end
+				if pos then
+					local num = tonumber(pos)
+					local combine = ""
+					for i = start+1, #input do
+						local line = input[i]
+						if line then
+							if string.sub(line, 1,1) ~="-" and line ~= "children are {" then
+								combine = combine .. line
+							elseif line == "children are {" then
+								combine = combine .. "$"
+								for j = i+3, #input do
+									local childline = input[j]
+									if line and string.sub(childline, 1, 1) ~= "}" then
+										combine = combine .. childline
+									else
+										break
+									end
+								end
+							else
+								break
+							end
+						end
+					end
+					if not double then
```

Comment:
Generally, I prefer to have `if x then else end` rather than `if not x then else end` and it just increases the cognitive load when reading the method and trying to understand what's going on here.